src/libstd/process.rs

```diff
@@ -828,13 +828,33 @@ impl Child {
 /// this function at a known point where there are no more destructors left
 /// to run.
 ///
+/// ## Platform-specific behavior
+///
+/// Despite `code` being an `i32`, some platforms take into account only a
+/// certain range of these bits. On macOS and Linux, `exit`'s underlying system
+/// function only considers the lower eight bits of `code`. For example, on
+/// these platforms, calling `exit` with values `0x00_00` and `0x0f_00` will be
+/// handled identically. Other platforms, like Windows, will take into account
+/// all 32 bits of `code`.
```

Comment:
Oh sorry to be clear that opengroup page is a Unix-only thing, which is basically just explaining why only 8 bits are transmitted to the parent on Unix.